ci/infra/testrunner/tests/test_node_reboot.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import json
+import time
+
+import pytest
+from timeout_decorator import timeout
+
+logger = logging.getLogger("testrunner")
+
+
+@timeout(120)
+def wait_for_node_reboot(node_type, node_number, platform):
+    while True:
+        try:
+            platform.ssh_run(node_type, node_number, 'echo hello')
+        except Exception as ex:
+            logger.debug(ex)
+        else:
+            break
+
+        time.sleep(5)
+
+
+@timeout(120)
+def check_nodes_ready(kubectl):
+    while True:
+        try:
+            kubectl.run_kubectl('wait --for=condition=Ready node --all')
+        except Exception as ex:
+            logger.debug(ex)
+        else:
+            break
+
+        time.sleep(5)
+
+
+@timeout(120)
+def check_pods_ready(kubectl):
+    while True:
+        try:
+            pods = json.loads(kubectl.run_kubectl('get pods --all-namespaces -o json'))['items']
+
+            for pod in pods:
+                pod_status = pod['status']['phase']
+                assert pod_status == 'Running', f'Pod {pod["metadata"]["name"]} status {pod_status} != Running'
```

Comment:
Not all pods have to be running, some can be completed - I think this will always fail on cilium-operator, or not?
Btw there is kubectl wait for pods too, but I am not sure if it's reliable.